charts/newrelic-infrastructure/templates/controlplane/daemonset.yaml

```diff
@@ -24,9 +24,7 @@ spec:
     metadata:
       annotations:
         checksum/nri-kubernetes: {{ include (print $.Template.BasePath "/controlplane/configmap.yaml") . | sha256sum }}
-        {{- if not (include "newrelic.licenseCustomSecretName" .) }}
-        checksum/secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
-        {{- end }}
+        checksum/license-secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
```

Comment:
is this ok? How this will work if there is not custom secret ? I saw previous one not rendering this annotation.

Reply:
> How this will work if there is not custom secret ? I saw previous one not rendering this annotation.

Reply:
Oh, sorry.

Example `values.yaml`:
```yaml
global:
  cluster: test
  customSecretName: test
  customSecretLicenseKey: test
```

`% helm template test . -f values-custom.yaml -s templates/kubelet/daemonset.yaml`
```yaml
apiVersion: apps/v1
kind: DaemonSet
spec:
  template:
    metadata:
      annotations:
        checksum/license-secret: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b
```
Empty secret, empty file, a hash of nothing. But hashed. Ready to redeploy if this changes.

Reply:
could you add a comment somewere? to me it could be confusing to have this in the annotations without having a license-secret.
I would prefer not to render it if the secret not exist. I guess that the redeploy will happen anyway if the license is changed since is an envar IRRC

Reply:
Fair point. [From #7bd684d](https://github.com/newrelic/nri-kubernetes/pull/383/commits/7bd684da4abd9bd632f5650288ccb4367b31b8b2) it only renders when a secret is rendered.

I am keeping this thread open as @roobre pointed out the security issue. I forgot to add that the hash is for the whole secret so it changes also when the chart updates because it includes the labels. Brute forcing a whole YAML, even if you know that it only changes a couple of things like version or release name is madness from my point of view.

Reply:
would make senes to add a random(like `{{ randAlphaNum 5 | quote }}`) label/annotation or extra secret to the secret  to make brut force even harder? 

But anyway I think the auto rolling feature is worth to be implemented. Unprivileged users should not have the permissions to see this annotations right?

Reply:
Even if they can we were using this before this refactor without hesitation. The brute force needed for this would be massive. All the labels are salt enough to keep the secret secret.